server/routes/functionalSkills/functionalSkillsController.ts

```diff
@@ -14,18 +18,50 @@ export default class FunctionalSkillsController {
       prisonNumber,
       req.user.username,
     )
-    const latestFunctionalSkillsFromCurious = mostRecentFunctionalSkills(functionalSkillsFromCurious)
-    const allFunctionalSkillsFromCurious = allFunctionalSkills(functionalSkillsFromCurious)
 
     const { problemRetrievingData } = functionalSkillsFromCurious
+
+    // Get all functional skills with the type 'ENGLISH' and add the prison name to the object for each from the prison service
     const englishSkills = functionalSkillsByType(functionalSkillsFromCurious.assessments, 'ENGLISH')
+    await Promise.all(
+      englishSkills.map(async assesment => {
+        const prison = await this.prisonService.lookupPrison(assesment.prisonId, req.user.username)
+        if (prison) {
+          // TODO refactor to avoid param-reassign eslint rule
+          // eslint-disable-next-line no-param-reassign
+          assesment.prisonName = prison.prisonName
+        }
+      }),
+    )
+
+    // Get all functional skills with the type 'MATHS' and add the prison name to the object for each from the prison service
     const mathsSkills = functionalSkillsByType(functionalSkillsFromCurious.assessments, 'MATHS')
+    await Promise.all(
+      mathsSkills.map(async assesment => {
+        const prison = await this.prisonService.lookupPrison(assesment.prisonId, req.user.username)
+        if (prison) {
+          // TODO refactor to avoid param-reassign eslint rule
+          // eslint-disable-next-line no-param-reassign
+          assesment.prisonName = prison.prisonName
+        }
+      }),
+    )
+
+    // Get all functional skills with the type 'DIGITAL_LITERACY' and add the prison name to the object for each from the prison service
     const digitalSkills = functionalSkillsByType(functionalSkillsFromCurious.assessments, 'DIGITAL_LITERACY')
+    await Promise.all(
```

Comment:
OK, we can tidy this up a bit by putting all the lookups in one `Promise.all` (we dont need 3 `Promise.all`s, each containing 1 promise! 😁 )
```
await Promise.all(
  englishSkills.map(async assesment => { ....
  }),
  mathsSkills.map(async assesment => { ....
  }),
  digitalSkills.map(async assesment => { ....
  }),
)

Reply:
Just noticed assessment is spelt wrong as well 😁
`async assesment` should be `async assessment`

🤣